File: solutions/go/word-count/1/word_count.go
// wordcount implements a package to countthe numbers of words.
package wordcount

import "strings"


type Frequency map[string]int
// function WordCount counts the number of times a word appears in a phrase
func WordCount(phrase string) Frequency {
	wc := make(Frequency)
	phrase = strings.ToLower(phrase)
	ephrase :=""
	for _,v := range phrase {
		if (string(v)>="a" && string(v)<="z")||
		   (string(v)>="0" && string(v)<="9")|| string(v)=="'" {
			ephrase = ephrase + string(v)
		} else {
			ephrase = ephrase + " "
		}
	}
	words := strings.Fields(ephrase)
	for _,w := range words {
		if string(w[0])=="'" {
			w=w[1:]
		}
		if string(w[len(w)-1:])=="'" {
			 w=w[:len(w)-1]
		}
		wc[w] += 1
	}
	return wc
}

File: solutions/cpp/word-count/1/word_count.cpp
#include "word_count.h"
using namespace std;
namespace word_count {
    const map<string, int> words(string phrase) {
        map<string, int> wc;
        int start_pos = 0;
        string word;
        while (start_pos < (int)phrase.length()){
            word = get_next_word(phrase, &start_pos);

            if (word != "") {
                if(wc.find(word) == wc.end()) {
                    wc[word]=1;
                   
                } else {
                    wc[word]++;
                    
                }
            }
        }
   
        return wc;
    }
    string get_next_word(string phrase, int* start_pos) {
        string ws = " ,.\n\t!&@$%^:";
        int pos;

        /* skip white spaces */
        for (pos = *start_pos; pos<(int)phrase.length(); pos++) {
            if (ws.find(phrase.substr(pos,1))>=phrase.length()) break;
        }
        
        /* get next word */
        string next_word = "";
        for(;pos<(int)phrase.length(); pos++){
            if (ws.find(phrase.substr(pos,1))<phrase.length()) break;
            next_word += phrase.substr(pos,1);
            
        }

        *start_pos = pos;
        string lc_next_word="";
        for(int i=0; i<(int)next_word.length(); i++) {
            lc_next_word+=tolower(next_word.at(i));
        }
        if (lc_next_word.substr(0,1)=="'")lc_next_word=lc_next_word.substr(1,lc_next_word.length()-2);
        //if (lc_next_word.substr(lc_next_word.length()-1,1)=="'")lc_next_word=lc_next_word.substr(0,lc_next_word.length()-1);
        return lc_next_word;
    }   
}  // namespace word_count
